src/bevent.c

```diff
@@ -324,6 +615,57 @@ void uag_event_unregister(ua_event_h *h)
 }
 
 
+/**
+ * Register an Event handler
+ *
+ * @param eh  Event handler
+ * @param arg Handler argument
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+int  bevent_register(bevent_h *eh, void *arg)
```

Comment:
Would it make sense to allow an event handler to only register for a certain event class? E.g. an event handler is only interested in UA events, or only CALL events (or CLASS_UNDEFINED for all events) and the handler is only called for those event types.

Reply:
Or allow flags? E.g. `BEVENT_CLASS_UA | BEVENT_CLASS_CALL`

Reply:
Good idea. Let's do this in a separate PR. Because `bevent_register()` currently is only used in the selftest, it is no problem to change the API function later. I add this to the TODO list  in the description.